internal/controller/predicates/predicates.go

```diff
@@ -33,6 +33,15 @@ func (GrafanaPodRecreatedPredicate) Update(e event.UpdateEvent) bool {
 	if !ok {
 		return false
 	}
+
+	if newPod == nil {
+		return false
+	}
+
+	if !newPod.DeletionTimestamp.IsZero() {
```

Comment:
Could comments improve readability?

```suggestion
        // do not trigger on pod deletion
	if !newPod.DeletionTimestamp.IsZero() {
```